File: shoestorefe/lib/domain/entities/receipt.dart
import 'receipt_detail.dart';

class Receipt {
  final int id;
  final String receiptNumber;
  final int supplierId;
  final String? supplierName;
  final int? storeId;
  final String? storeName;
  final int createdBy;
  final String? creatorName;
  final int statusId;
  final DateTime createdAt;
  final DateTime? receivedDate;
  final double totalAmount;
  final List<ReceiptDetail> details;

  Receipt({
    required this.id,
    required this.receiptNumber,
    required this.supplierId,
    this.supplierName,
    this.storeId,
    this.storeName,
    required this.createdBy,
    this.creatorName,
    required this.statusId,
    required this.createdAt,
    this.receivedDate,
    required this.totalAmount,
    required this.details,
  });
}

File: ShoeStore/ShoeStore.Domain/Entities/Receipt.cs
namespace ShoeStore.Domain.Entities;

public class Receipt
{
    public long Id { get; set; }
    public string ReceiptNumber { get; set; } = null!;
    public int SupplierId { get; set; }
    public long CreatedBy { get; set; }
    public int? StoreId { get; set; }
    public int StatusId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? ReceivedDate { get; set; }
    public decimal TotalAmount { get; set; }

    public Supplier Supplier { get; set; } = null!;
    public User Creator { get; set; } = null!;
    public Store? Store { get; set; }
    public Status Status { get; set; } = null!;
    public ICollection<ReceiptDetail>? ReceiptDetails { get; set; }
}

